modules/globebrowsing/tile/rawtiledatareader/rawtiledatareader.cpp

```diff
@@ -0,0 +1,416 @@
+                // +-+-----+               |              | <-- io.read.fullRegion  
+                //                         |              |
+                //                         +--------------+
+
+                // Example:
+                // The cutoff region has been repeated along one of its sides, but 
+                // as we can see in this example, it still has a top part outside the
+                // defined gdal region. This is handled through recursion.
+                RawTile::ReadError err = repeatedRasterRead(
+                    rasterBand, cutoff, dataDestination, depth + 1);
+
+                worstError = std::max(worstError, err);
+            }
+        }
+    }
+        
+    RawTile::ReadError err = rasterRead(rasterBand, io, dataDestination);
+
+    // The return error from a repeated rasterRead is ONLY based on the main region,
+    // which in the usual case will cover the main area of the patch anyway
+    return err;
+}
+
+std::shared_ptr<TileMetaData> RawTileDataReader::getTileMetaData(
+    std::shared_ptr<RawTile> rawTile, const PixelRegion& region)
+{
+    ensureInitialized();
+    size_t bytesPerLine = _dataLayout.bytesPerPixel * region.numPixels.x;
+
+    TileMetaData* preprocessData = new TileMetaData();
+    preprocessData->maxValues.resize(_dataLayout.numRasters);
+    preprocessData->minValues.resize(_dataLayout.numRasters);
+    preprocessData->hasMissingData.resize(_dataLayout.numRasters);
+        
+    std::vector<float> noDataValues;
+    noDataValues.resize(_dataLayout.numRasters);
+
+    for (size_t c = 0; c < _dataLayout.numRasters; c++) {
+        preprocessData->maxValues[c] = -FLT_MAX;
+        preprocessData->minValues[c] = FLT_MAX;
+        preprocessData->hasMissingData[c] = false;
+        noDataValues[c] = noDataValueAsFloat();
+    }
+
+    for (size_t y = 0; y < region.numPixels.y; y++) {
+        size_t yi = (region.numPixels.y - 1 - y) * bytesPerLine;
+        size_t i = 0;
+        for (size_t x = 0; x < region.numPixels.x; x++) {
+            for (size_t c = 0; c < _dataLayout.numRasters; c++) {
+                float noDataValue = noDataValueAsFloat();
+                float val = tiledatatype::interpretFloat(
+                    _dataLayout.glType,
+                    &(rawTile->imageData[yi + i])
+                );
+                if (val != noDataValue) {
+                    preprocessData->maxValues[c] = std::max(
+                        val,
+                        preprocessData->maxValues[c]
+                    );
+                    preprocessData->minValues[c] = std::min(
+                        val,
+                        preprocessData->minValues[c]
+                    );
+                }
+                else {
+                    preprocessData->hasMissingData[c] = true;
+                }
+                i += _dataLayout.bytesPerDatum;
+            }
+        }
+    }
+
+    for (size_t c = 0; c < _dataLayout.numRasters; c++) {
```

Comment:
Changed to `raster` instead of `c`. More explicit. `i` was already in use. More consistent to use `raster` everywhere than to mix in `i` for some loops